src/App.tsx

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { LoadingError } from './components/LoaderError/LoadingError';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoading(true);
+
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setHasLoadingError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    let filteredTodos = [...todos];
+
+    if (query) {
+      filteredTodos = filteredTodos.filter(todo => (
+        todo.title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+      ));
+    }
+
+    switch (filterBy) {
+      case FilterBy.Completed:
+        filteredTodos = filteredTodos.filter(todo => todo.completed);
+        break;
+
+      case FilterBy.Active:
+        filteredTodos = filteredTodos.filter(todo => !todo.completed);
+        break;
+
+      default:
+        break;
+    }
+
```

Comment:
Remove the `loading` dependency from the useMemo hook on line 66, as it is not used inside the function. This will prevent unnecessary re-computations 🚀.

Reply:
I need it to first todos loading

Reply:
hmm, it seems to be not good, since your function shouldn't be dependent on external data to avoid side effects